File: php/is_prime.php
<?php
/**
 * This script will determine if a number is a prime number, run it as so 'php is_prime.php 7'
 */

function is_prime($number) {
  if($number == 2) {
    print 'Prime';
    return;
  }

  if($number <= 1 || ($number % 2) == 0) {
    print 'Not a prime';
    return;
  }

  for($i = 3; $i <= ceil(sqrt($number)); $i++) {
    if(($number % $i) == 0) {
      print 'Not a prime';
      return;
    }
  }

  print 'Prime';
}

if(!isset($argv[1])) {
  print 'Please specify a number as the only parameter, i.e. php is_prime 7';
  exit();
}

is_prime($argv[1]);

File: cpp/is_prime.cpp
/**
 * This program will determine if a number is a prime number, to run:
 * g++ is_prime.cpp -o is_prime
 * ./is_prime 7
 */
#include <iostream>
#include <math.h>
using namespace std;

int main(int argc, char *argv[]) {
  if(argc == 1) {
    cout << "No number passed in as paramter" << endl;
  }

  int number = *argv[1] - '0';

  if(number == 2) {
    cout << "Prime";
    return 0;
  }

  if(number <= 1 || (number % 2) == 0) {
    cout << "Not a prime";
    return 0;
  }

  for(int i = 1; i <= ceil(sqrt(number)); i++) {
    if((number % i) == 0) {
      cout << "Not a prime";
      return 0;
    }
  }

  cout << "Prime";
  return 0;
}
